rasa/utils/tensorflow/rasa_layers.py

```diff
@@ -30,7 +31,165 @@
+                    reg_lambda=reg_lambda,
+                )
+            elif isinstance(layer, layers.DenseForSparse):
+                attribute = layer.get_attribute()
+                feature_type = layer.get_feature_type()
+                if (
+                    attribute in new_sparse_feature_sizes
+                    and feature_type in new_sparse_feature_sizes[attribute]
+                ):
+                    new_feature_sizes = new_sparse_feature_sizes[attribute][
+                        feature_type
+                    ]
+                    old_feature_sizes = old_sparse_feature_sizes[attribute][
+                        feature_type
+                    ]
+                    self._check_sparse_feature_decreased_sizes(
+                        new_sparse_feature_sizes=new_feature_sizes,
+                        old_sparse_feature_sizes=old_feature_sizes,
+                    )
+                    if sum(new_feature_sizes) > sum(old_feature_sizes):
+                        new_layer = self._create_dense_for_sparse_layer(
+                            dense_layer=layer,
+                            new_sparse_feature_sizes=new_feature_sizes,
+                            old_sparse_feature_sizes=old_feature_sizes,
+                            attribute=attribute,
+                            feature_type=feature_type,
+                            reg_lambda=reg_lambda,
+                        )
+                        self._tf_layers[name] = new_layer
+
+    @staticmethod
+    def _check_sparse_feature_decreased_sizes(
+        new_sparse_feature_sizes: List[int], old_sparse_feature_sizes: List[int],
+    ) -> None:
+        """Checks if the sizes of sparse features are decreased during fine-tuning.
+
+        Sparse feature sizes might decrease after changing the nlu file. This can
+        happen for example with `LexicalSyntacticFeaturizer`. We don't support
+        this behaviour and we raise a warning if this happens.
+
+        Args:
+            new_sparse_feature_sizes: sizes of current sparse features for a
+                                      specific attribute and feature type.
+            old_sparse_feature_sizes: sizes of sparse features the model was trained on
+                                      before for a specific attribute and feature type.
+        """
+        for new_size, old_size in zip(
+            new_sparse_feature_sizes, old_sparse_feature_sizes
+        ):
+            if new_size < old_size:
+                raise Exception(
+                    "Sparse feature sizes are decreased."
+                    "The NLU file was changed in a way that caused removing some"
+                    "features from certain featurizers (This might have been "
+                    "`LexicalSyntacticFeaturizer`). We don't support this behaviour"
+                    "at this point. One possible way to avoid this "
+                    "warning is retraining the model from scratch."
```

Comment:
```suggestion
                    "Sparse feature sizes have decreased from the last time training was run."
                    "The training data was changed in a way that resulted in some"
                    "features not being present in the data anymore. This can happen if you had "
                    "`LexicalSyntacticFeaturizer` in your pipeline. The pipeline cannot support incremental training in this setting. We recommend you to retrain the model from scratch."
```